homeassistant/components/shelly/__init__.py

```diff
@@ -298,15 +304,15 @@ async def async_setup(self):
         self.device_id = entry.id
         self.device.subscribe_updates(self.async_set_updated_data)
 
-    def shutdown(self):
+    def shutdown(self) -> None:
         """Shutdown the wrapper."""
         if self.device:
             self.device.shutdown()
             self._async_remove_device_updates_handler()
             self.device = None
 
     @callback
-    def _handle_ha_stop(self, _):
+    def _handle_ha_stop(self, _: Any) -> None:
```

Comment:
This is not type any, but a event.

Reply:
I'd also add a name for clarity.
```suggestion
    def _handle_ha_stop(self, _event: Event) -> None:
```

Reply:
Of course you're right. Fixed.